### ISSUE TITLE
Cloud Usage graph shows 3 times the actual value

### ISSUE THREAD

**Issue by PreNoob** [cloud-carbon-footprint/cloud-carbon-footprint#1281]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/trunk/CONTRIBUTING.md) for this project.
- [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

- **Container Images:**
`cloudcarbonfootprint/api:release-2023-10-17` and `cloudcarbonfootprint/client:release-2023-10-17`
- **Cloud Provider:**
AWS (multiple accounts) 
- **Operating System:**
N/A
- **Browser:**
N/A

### Expected Behavior
The Cloud Usage graph should display a cost values similar to the one from AWS cost explorer.

### Actual Behavior
When checking the Cloud Usage graph, it display 2-3 times the actual cost value displayed in the cost explorer. 

### To Reproduce
Sadly, I can't reproduce this behavior deterministically. 

### Screenshots
N/A

### Additional Information
I've curled the endpoint `footprint?start=2022-12-15&end=2023-12-15&ignoreCache=false&groupBy=day&limit=50000&skip=0` (saw this endpoint in the Browser Dev Tools), filtered by account using `jq "[ .[] | .serviceEstimates |= map(select(.accountId == \"<ACCOUNT_ID>\")) ]"` and used the following python 3 script to calculate the cost per day.

```python
import json
from datetime import datetime
import sys

def calculate_cost_per_day(data):
    results = []
    for entry in data:
        cost_per_day = 0
        for service in entry['serviceEstimates']:
            cost_per_day += service['cost']
        timestamp = entry['timestamp']
        formatted_timestamp = datetime.strptime(timestamp, "%Y-%m-%dT%H:%M:%S.%fZ").strftime("%Y-%m-%d")
        results.append({
            "timestamp": formatted_timestamp,
            "cost_per_day": cost_per_day
        })
    return results

def main(filename):
    with open(filename, 'r') as file:
        data = json.load(file)

    results = calculate_cost_per_day(data)

    output_json = json.dumps(results, indent=2)
    print(output_json)

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: python cost_per_day.py filename")
    else:
        filename = sys.argv[1]
        main(filename)
```
The output values of that script are at least in the same ballpark as the value from AWS Cost explorer. Therefore I assume there must be an issue with the client itself. 

© 2021 Thoughtworks, Inc.

**Comment by 4upz** [comment]
Thanks for opening this issue @PreNoob, I was able to replicate this behavior and it is indeed an issue with the client. There seems to be a bug with the pagination logic where it sometimes makes reads the same estimates twice. This causes for duplicate numbers to be shown in the graph, hence the 2x-3x totals you're seeing.

I am currently working on a fix for this to be included in the next release

**Comment by 4upz** [comment]
@PreNoob This should be fixed with 9346ca9d668f3a39ca7279fe39dc3fb607bbd387 and is queued for the next release. Feel free to pull down the changes or keep an eye out for the next release, and let us know if you see the issue still persisting.

Closing this issue now, but happy to reopen or assist if you have more concerns
